test/schema_test.exs

```diff
@@ -36,38 +36,37 @@ defmodule ArcTest.Ecto.Schema do
 
   test "supports :invalid changeset" do
     cs = TestUser.changeset(%TestUser{})
-    assert cs.valid? == false
-    assert cs.changes == %{}
-    assert cs.required == [:avatar]
+    assert cs.valid?   == false
+    assert cs.changes  == %{}
+    assert cs.errors   == [avatar: {"can't be blank", []}]
   end
 
   test_with_mock "cascades storage success into a valid change", DummyDefinition, [store: fn({"/path/to/my/file.png", %TestUser{}}) -> {:ok, "file.png"} end] do
     cs = TestUser.changeset(%TestUser{}, %{"avatar" => "/path/to/my/file.png"})
     assert called DummyDefinition.store({"/path/to/my/file.png", %TestUser{}})
     assert cs.valid?
-    %{file_name: "file.png", updated_at: updated_at} = cs.changes.avatar
-    assert updated_at = %Ecto.DateTime{}
+    %{file_name: "file.png", updated_at: _} = cs.changes.avatar
   end
 
   test_with_mock "cascades storage error into an error", DummyDefinition, [store: fn({"/path/to/my/file.png", %TestUser{}}) -> {:error, :invalid_file} end] do
     cs = TestUser.changeset(%TestUser{}, %{"avatar" => "/path/to/my/file.png"})
     assert called DummyDefinition.store({"/path/to/my/file.png", %TestUser{}})
     assert cs.valid? == false
-    assert cs.errors == [avatar: {"can't be blank", []}, avatar: {"is invalid", [type: ArcTest.Ecto.Schema.DummyDefinition.Type]}]
+    assert cs.errors == [avatar: {"is invalid", [type: ArcTest.Ecto.Schema.DummyDefinition.Type]}]
```

Comment:
according to [docs](https://hexdocs.pm/ecto/Ecto.Changeset.html#validate_required/3): `Note the error won’t be added though if the field already has an error.`